packages/graphiql-plugin-utils/src/vite-plugin.ts

```diff
@@ -0,0 +1,140 @@
+import react from '@vitejs/plugin-react';
+import svgr from 'vite-plugin-svgr';
+import postCssNestingPlugin from 'postcss-nesting';
+import dts from 'vite-plugin-dts';
+import type { Plugin, UserConfig } from 'vite';
+
+const IS_UMD = process.env.UMD === 'true';
+
+type PluginOptions = {
+  /**
+   * The plugin name used by the UMD global
+   */
+  umdExportName: string;
+  /**
+   * The name of the plugin exported by the umd global
+   * Defaults to a normalized version of the package.json `name` field
+   */
+  fileName?: string;
+};
+
+const normalizePackageName = (name: string): string => {
+  return name.replace('@', '').replace('/', '-');
+};
+
+function graphiqlPluginConfig({
+  umdExportName,
+  fileName,
+}: PluginOptions): Plugin {
+  return {
+    name: 'vite-plugin-graphiql-plugin:build',
+    enforce: 'pre',
+    async config(config) {
+      let packageJSON;
+      // TODO: process.env.npm_package_json_path is no longer present in vite. study
+      try {
+        // more vite plugins to see how others load the full package.json
+        packageJSON = await import(`${process.env.PWD}/package.json`);
+      } catch {
+        throw new Error(
+          'the graphiql plugin vite plugin currently only works if you execute vite commands from the same directory as package.json',
+        );
+      }
+      const external = [];
+      const userExternal = config.build?.rollupOptions?.external;
+      if (userExternal) {
+        if (Array.isArray(userExternal)) {
+          external.push(...userExternal);
+        } else if (typeof userExternal === 'string') {
+          external.push(userExternal);
+        }
+      }
+      if (typeof config.css?.postcss === 'string') {
+        throw new Error('config.css.postcss passed as string, must be object');
+      }
+      return {
+        ...config,
+        css: {
+          ...config?.css,
+          postcss: {
+            ...config?.css?.postcss,
+            plugins: [
+              ...(config?.css?.postcss?.plugins ?? []),
+              postCssNestingPlugin(),
+            ],
+          },
+        },
+        esbuild: {
+          ...config?.esbuild,
+          // We use function names for generating readable error messages, so we want
+          // them to be preserved when building and minifying.
+          keepNames: true,
+        },
+        build: {
+          ...config?.build,
+          emptyOutDir: !IS_UMD,
+          lib: {
+            ...config?.build?.lib,
+            entry: 'src/index.tsx',
+            fileName: IS_UMD
+              ? fileName ?? normalizePackageName(packageJSON.name)
+              : 'index',
```

Comment:
we can stay with `index` name to simplify things, no need additional configuration
```suggestion
            fileName: 'index',
```

Reply:
@B2o5T you remember the recent breakage where users asked for the named export file yes?

Reply:
It was fixed, there is no need to configure the output filename, keep it as `index` for simplifying things, no need to add additional options

Reply:
this eliminates the need for the script you wrote as the workaround

Reply:
it's fine to introduce breaking change in `0.x`

Reply:
@B2o5T it is an unnecessary breaking change, especially given that users just asked us to fix it, and most unpkg users are using latest. we should only introduce breaking changes if there is a purpose

Reply:
it's a breaking change for a valid semver version, users asked to fix for patch version - I fixed it

> and this is just a convention you prefer

it's not my preference, most packages use `index` as the name for an entry point, here

`https://unpkg.com/@graphiql/plugin-explorer@0.1.12/dist/graphiql-plugin-explorer.umd.js`

there is no need to repeat `graphiql-plugin-explorer`, you can already see `@graphiql/plugin-explorer` as a package name